src/protocol/vm/adapter_contract.rs

```diff
@@ -105,15 +105,15 @@ where
                         .first()
                         .cloned()
                         .ok_or_else(|| {
-                            SimulationError::DecodingError("There wasn't a calculated price".into())
+                            SimulationError::FatalError("There wasn't a calculated price".into())
                         })?;
 
                     Ok((received_amount, gas_used, price))
                 }
-                _ => Err(SimulationError::DecodingError("Incorrect types found for price".into())),
+                _ => Err(SimulationError::FatalError("Incorrect types found for price".into())),
```

Comment:
This means the adapter contract does not respect the interface right? Maybe the error message can be improved here by also returning the received types vs expected ones.